### PR TITLE
Support Ray 2.0

### PR DESCRIPTION
**Pull request by amogkam**
Signed-off-by: Amog Kamsetty <amogkamsetty@yahoo.com>

Remove usage of `ray.train.utils.get_address_and_port` to support recent changes in Ray 2.0

Closes https://github.com/pytorch/torchx/issues/584

Test plan:

### CODE REVIEW

**Review by msaroufim — APPROVED** [review]

**Comment by amogkam** [comment]
Is this ready to be merged in?

**Comment by msaroufim** [comment]
Seems like #587 also includes a Ray 2.0 fix? I'll let @d4l3k decide what to merge

**Comment by d4l3k** [comment]
The CI is failing on both of them which is why I haven't merged -- #587 is a bit closer to being ready to land

File: torchx/schedulers/ray/ray_driver.py [hunk 1]
```diff
@@ -95,7 +95,12 @@ def schedule(self, actor_id: str) -> CommandActorScheduled:
         return CommandActorScheduled(actor_id)
 
     def get_actor_address_and_port(self) -> Tuple[str, int]:
-        return get_address_and_port()
+        addr = ray.util.get_node_ip_address()
+        with closing(socket.socket(socket.AF_INET, socket.SOCK_STREAM)) as s:
```
Comment (kurman) [inline torchx/schedulers/ray/ray_driver.py:99]: Did you mean to use https://docs.python.org/3/library/contextlib.html#contextlib.closing  here?

**Review by kurman — COMMENTED** [review]
@amogkam I am trying merge the changes, there are test failures due to missing module. Do you mind looking at the comment?

**Pull request closed without merging**